dora/active_sampling/lin_sampler.py

```diff
@@ -0,0 +1,391 @@
+        """Predict the query mean and standard deviation.
+
+        Parameters
+        ----------
+        Xq: numpy.ndarray
+           query points.
+        real: bool, optional
+        Returns
+        -------
+        Yq_mean: numpy.ndarray
+                mean of the prediction at query points.
+        Yq_std: numpy.ndarray
+                standard deviation of the prediction at query points.
+        """
+        assert self.hyperparams, "Sampler is not trained yet."
+        if real:
+            X_real, y_real = self.get_real_data()
+            regressors = [self._compute_AmuL(
+                self.hyperparams[i_task], self.FeatureGen(X_real), y_real[:, i_task]) for i_task in range(self.n_tasks)]
+        else:
+            regressors = self.regressors
+
+        thetaQuery = self.FeatureGen(Xq)
+        Yq_mean = np.array([self._compute_posterior(r, thetaQuery)[0] for r in regressors]).T
+        Yq_std = np.array([self._compute_posterior(r, thetaQuery)[1] for r in regressors]).T
+        return Yq_mean, Yq_std
+
+
+    def get_real_data(self):
+        """Obtain the observed data.
+
+        Returns
+        -------
+        self.X()[real_flag]: numpy.ndarray
+                            the observed feature locations.
+        self.y()[real_flag]: numpy.ndarray
+                            the observed target outputs.
+        """
+        assert self.X
+        assert self.y
+        real_flag = ~self.virtual_flag()
+        return self.X()[real_flag], self.y()[real_flag]
+
+
+    def eval_acq(self, Xq):
+        """Evaluate the acquisition function.
+
+        Parameters
+        ----------
+        Xq: numpy.ndarray
+            query points.
+        Returns
+        -------
+        yq_acq: numpy.ndarray
+                acquisition function value at Xq.
+        np.argmax(yq_acq): int
+                          index of the maximum acquisition function value.
+        """
+        if len(Xq.shape)==1:
+            Xq = Xq[:, np.newaxis]
+        self.update_y_mean()
+        thetaQuery = self.FeatureGen(Xq)
+        Yq_mean = np.array([self._compute_posterior(r, thetaQuery)[0] for r in self.regressors]).T
+        Yq_std = np.array([self._compute_posterior(r, thetaQuery)[1] for r in self.regressors]).T
+        acq_defs_current = acq_defs(y_mean=self.y_mean,
+                                    explore_priority=self.explore_priority)
+        yq_acq = acq_defs_current[self.acq_name](Yq_mean, Yq_std**2)
+        return yq_acq, np.argmax(yq_acq)
+
+
+def basis_defs(mu, s):
+    """Basis function definitions.
+
+    Parameters
+    ----------
+    mu: numpy.ndarray
+        mean of the basis function.
+    s: float
+       spatial width of the basis function.
+    Returns
+    -------
+    dict
+    dictionary of basis functions.
+    """
+    #TODO: add some other basis functions
+    return{
+        'radial': lambda x: np.exp(-(((x - mu.transpose())**2)/(2*s**2))) }
+
+
+def acq_defs(y_mean=0, explore_priority=1.):
```

Comment:
If this is a copy and paste from the GP class, it probably makes sense to have an `acq_defs.py` that they both call. I'll add an issue for it and someone can address it later.